Extension/src/LanguageServer/client.ts

```diff
@@ -1536,7 +1556,7 @@ export class DefaultClient implements Client {
             resetDatabase: resetDatabase,
             edgeMessagesDirectory: path.join(util.getExtensionFilePath("bin"), "messages", getLocaleId()),
             localizedStrings: localizedStrings,
-            settings: this.getAllSettings()
+            settings: await this.getAllSettings()
```

Comment:
I'm almost certain we avoided asyncs early in client creation code, as a fix for issues that occurs due to race conditions at startup.  I don't recall the exact details at the moment.  But the call to `createLanguageClient` is not awaited (it's called from a constructor, and constructors aren't async).  So, with your change, it's possible for the client to get constructed and presumably calls to `this.languageClient` to occur before that variable is properly set.  I think that's why there are no awaits there until after `languageClient` is set and everything has been hooked up.

Would it be possible to remove this async/await and deliver the setting value without awaiting to see if the flight is enabled?  Then, gate the functionality later based on whether the flight is enabled?